test/query.js

```diff
@@ -16,14 +17,64 @@ const offsetApplied = require('./fixtures/offset-applied.json')
 
 // const moment = require('moment')
 
-describe('Query operatons', () => {
+describe('Query operations', () => {
+  it('should return the expected response schema for an optionless query', () => {
+    const response = FeatureServer.query(data, {})
+
+    const schema = Joi.object().keys({
+      'objectIdFieldName': Joi.string().valid('OBJECTID'),
+      'globalIdFieldName': Joi.string().allow(''),
+      'hasZ': Joi.boolean().valid(false),
+      'hasM': Joi.boolean().valid(false),
+      'geometryType': Joi.string().valid('esriGeometryPoint', 'esriGeometryLine', 'esriGeometryPolygon'),
+      'spatialReference': Joi.object().keys({
+        'wkid': Joi.number().integer().min(0)
+      }),
+      'fields': Joi.array().min(1).items(Joi.object().keys({
+        'name': Joi.string().when('type', {is: 'esriFieldTypeOID', then: Joi.valid('OBJECTID'), otherwise: Joi.string()}),
+        'type': Joi.string().allow('esriFieldTypeOID', 'esriFieldTypeInteger', 'esriFieldTypeDouble', 'esriFieldTypeString', 'esriFieldTypeDate'),
+        'alias': Joi.string().when('type', {is: 'esriFieldTypeOID', then: Joi.valid('OBJECTID'), otherwise: Joi.string()}),
+        'length': Joi.optional().when('type', {
+          is: Joi.string().allow('esriFieldTypeString', 'esriFieldTypeDate'),
+          then: Joi.number().integer().min(0)
+        }),
+        'defaultValue': Joi.any().valid(null),
+        'domain': Joi.any().valid(null),
+        'sqlType': Joi.string().valid('sqlTypeOther', 'sqlTypeDouble')
+      })),
+      'features': Joi.array().items(Joi.object().keys({
+        'attributes': Joi.object().keys({
+          'OBJECTID': Joi.number().integer()
+        }).unknown(),
```

Comment:
is `unknown` set to `false` by default?  i can't remember.  what's the effect of calling `unknown` without any parameters?

Reply:
What I am going for here is that the `attributes` object must have a property `OBJECTID`, and it is allowed to have additional properties that I'm not concerned about.  Without the `unknown()`, the test fails because `attributes` has several other properties (in this case).  https://github.com/hapijs/joi/blob/master/API.md#objectunknownallow

Reply:
Ah, makes sense!